pretyping/evarconv.ml

```diff
@@ -380,12 +441,18 @@ let rec ise_stack2 no_app env evd f sk1 sk2 =
     match revsk1, revsk2 with
     | [], [] -> None, Success i
     | Stack.Case cse1 :: q1, Stack.Case cse2 :: q2 ->
-      let (t1, c1) = Stack.expand_case env evd cse1 in
-      let (t2, c2) = Stack.expand_case env evd cse2 in
+      let (ci1, u1, pms1, t1, br1) = Stack.expand_case env evd cse1 in
+      let (ci2, u2, pms2, t2, br2) = Stack.expand_case env evd cse2 in
+      let hd1 = mkIndU (ci1.ci_ind, u1) in
+      let hd2 = mkIndU (ci2.ci_ind, u2) in
+      let fctx i (ctx1, t1) (_ctx2, t2) = f (push_rel_context ctx1 env) i CONV t1 t2 in
       begin
         match ise_and i [
-          (fun i -> f env i CONV t1 t2);
-          (fun i -> ise_array2 i (fun ii -> f env ii CONV) c1 c2)]
+          (fun i -> compare_heads env i ~nargs:None hd1 hd2);
```

Comment:
Is `None` correct here?

Reply:
Yes, we know that the inductive is always fully applied in the return clause.

Reply:
oh None means "fully applied"
That's kinda weird
How about having `type nargs = FullyApplied | SomeArgs of int`?

Reply:
This API is not exported, I don't think it's worthwhile to introduce a new type for this.

Reply:
The current usage of option is misleading IMO, please make a new type.